spec/services/publish_service_spec.rb

```diff
@@ -49,6 +52,78 @@
       PublishService.call(current_edition, user, with_review: true)
     end
 
+    context "when the edition is not associated with a government" do
+      let(:government) { build(:government) }
+      let(:backdate) { Time.zone.parse("2018-11-11 10:00") }
+      let(:first_published_at) { Time.zone.parse("2019-11-11 10:00") }
+
+      it "tries to associate a government with edition backdate" do
+        edition = create(:edition,
+                         backdated_to: backdate,
+                         first_published_at: first_published_at)
+        expect(Government).to receive(:for_date).with(backdate)
+                          .and_return(government)
+
+        expect { PublishService.call(edition, user, with_review: true) }
+          .to change { edition.government_id }
+          .to(government.content_id)
+      end
+
+      it "tries to associate a government with first published date if there is no backdate" do
+        edition = create(:edition,
+                         backdated_to: nil,
+                         first_published_at: first_published_at)
+        expect(Government).to receive(:for_date).with(first_published_at)
+                          .and_return(government)
+
+        expect { PublishService.call(edition, user, with_review: true) }
+          .to change { edition.government_id }
+          .to(government.content_id)
+      end
+
+      it "Associate with current government as a fallback" do
```

Comment:
Nitpick: This test name should be "associates...".

Reply:
Thanks